### PR TITLE
fix: Leaderboard UI improvements

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
## Improvements:
Implemented the following things:
 - Added RegEx-sensitive search bar
 - Reorganized model and benchmark criteria in a way that's more sensible
 - Table improvements:
     - Better column names
     - Reorganized columns
     - Added clickable model links
     - Adjusted column widths

## Screenshot:
image

### CODE REVIEW

**Comment by KennethEnevoldsen** [comment]
Would change:
"should be sentenceTransformer compatible" -> "SentenceTransformer": from the context it is clear

Discussion points:
"Availability": Why not have it be "Open", or "Proprietary" and just check off both by default? Same for instructions.

**Review by isaac-chung — COMMENTED** [review]
In general, I don't see why we are changing some of the components unless they are based on collected feedback.

1. Search: Is it possible to replicate the current leaderboard search i.e. separate multiple queries with `;`? If not, I'd prefer to retain that instead of replacing it with regex.
2. Defaults: Current leaderboard has both "open" and "proprietary" selected, and both with/ without instructions selected. 
3. Benchmark: What is the expected use of the filters underneath "Benchmarks"? Are users expected to select a benchmark, _then_ add more filters if desired?

**Comment by x-tabdeveloping** [comment]
1. Using the ; syntax in the search bar is technically equivalent to using | in RegEx. And the current leaderboard technically also uses regex, so I think it's quite redundant.
2. I thought the way it was organized in the old leaderboard was a bit unclear, and I wanted to spell it out a bit more to communicate the intentions of these buttons clearer. I'm open to changing it up if people see fit.
3. The idea is that people can customize a benchmark to their liking. For instance if you are interested in retrieval and reranking in Danish but nothing else, you can select these and rank the models based on that. I think this is a rather useful thing to have.

**Comment by isaac-chung** [comment]
1. Great! Let's spell that out then.
2. Not sure how a difference in defaults help spell things out. Could we follow the current defaults?
3. Great! Definitely useful to have.

**Comment by isaac-chung** [comment]
1. Great! Let's spell that out then.
2. Not sure how a difference in defaults help spell things out. Could we follow the current defaults?
3. Great! Definitely useful to have.

**Review by isaac-chung — APPROVED** [review]
Looks good! A few small suggestions.

File: mteb/leaderboard/app.py [hunk 1]
```diff
@@ -117,15 +120,39 @@ def update_task_info(task_names: str) -> str:
...
                 """
             ### Model Selection
             Select models to rank based on an assortment of criteria. 
             """,
             )
             with gr.Group():
-                with gr.Row(elem_classes="overflow-y-scroll max-h-80"):
+                searchbar = gr.Textbox(
+                    label="Search Models",
+                    info="Search models by name (RegEx sensitive)",
```
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply mteb/leaderboard/app.py:152]: Good idea!!

**Pull request merged**